saleor/graphql/webhook/subscription_payload.py

```diff
@@ -0,0 +1,131 @@
+from typing import Any, Dict, Optional
+
+from celery.utils.log import get_task_logger
+from django.core.handlers.base import BaseHandler
+from django.http import HttpRequest
+from django.test.client import RequestFactory
+from graphql import GraphQLDocument, get_default_backend, parse
+from graphql.error import GraphQLSyntaxError
+from graphql.language.ast import FragmentDefinition, OperationDefinition
+from promise import Promise
+
+from ...app.models import App
+
+logger = get_task_logger(__name__)
+
+
+def validate_subscription_query(query: str) -> bool:
+    from ..api import schema
+
+    graphql_backend = get_default_backend()
+    try:
+        document = graphql_backend.document_from_string(schema, query)
+    except (ValueError, GraphQLSyntaxError):
+        return False
+    if not check_document_is_single_subscription(document):
+        return False
+    return True
+
+
+def check_document_is_single_subscription(document: GraphQLDocument) -> bool:
+    """Check if document contains only a single subscription definition.
+
+    Only fragments and single subscription definition are allowed.
+    """
+    result = True
+    subscriptions = []
+    for definition in document.document_ast.definitions:
+        if isinstance(definition, FragmentDefinition):
+            pass
+        elif isinstance(definition, OperationDefinition):
+            if definition.operation == "subscription":
+                subscriptions.append(definition)
+            else:
+                return False
+        else:
+            return False
+    if len(subscriptions) != 1:
+        result = False
+    return result
```

Comment:
`result` var introduces more complexity than it's worth imo.
```suggestion
    subscriptions = []
    for definition in document.document_ast.definitions:
        if isinstance(definition, FragmentDefinition):
            pass
        elif isinstance(definition, OperationDefinition):
            if definition.operation == "subscription":
                subscriptions.append(definition)
            else:
                return False
        else:
            return False

    return len(subscriptions) == 1
```

Reply:
nice, I like it :) fixed